### PR TITLE
FEInterfaceValues: pass to the constructor hp::QCollection

### PR DESCRIPTION
**Pull request by peterrum**
Similar to #10764 but for `FEInterfaceValues`. Motivation is to use the class for wedges and pyramids.

~Depends on #10764.~

### CODE REVIEW

File: include/deal.II/fe/fe_interface_values.h [hunk 1]
```diff
@@ -551,6 +581,12 @@ FEInterfaceValues<dim, spacedim>::reinit(
       fe_face_values_neighbor = &internal_fe_subface_values_neighbor;
     }
 
+  AssertDimension(fe_face_values->n_quadrature_points,
+                  fe_face_values_neighbor->n_quadrature_points);
+
+  const_cast<unsigned int &>(this->n_quadrature_points) =
```
Comment (drwells) [inline include/deal.II/fe/fe_interface_values.h:587]: Could we make `FEInterfaceValues::n_quadrature_points` non-const?

Comment (peterrum) [reply include/deal.II/fe/fe_interface_values.h:587]: Different reviewers different opinions: https://github.com/dealii/dealii/pull/10764#discussion_r541937876

Comment (drwells) [reply include/deal.II/fe/fe_interface_values.h:587]: Lets keep it this way for now but we should eventually clean this up.

**Review by drwells — APPROVED** [review]

**Pull request merged**